### ISSUE TITLE
negative generator loss

### ISSUE THREAD

**Issue by foobar8675** [lucidrains/stylegan2-pytorch#190]
First, I have to say I've learned a lot from working through your code.

I was hoping you might be able to give me some insight on an issue - sometimes, my generator has a negative loss when training, which as far as I understand, should not be possible. 

I was wondering if you have any insights as to why this might be? 

also, I notice for the generator loss, you are taking the mean of the outputs from the discriminator 
https://github.com/lucidrains/stylegan2-pytorch/blob/25ccfcfe1b5ca0fdd2c6b16bfb8df43180dfe09a/stylegan2_pytorch/stylegan2_pytorch.py#L994

where the nvlabs implementation uses a negative of the softplus
https://github.com/NVlabs/stylegan2-ada/blob/main/training/loss.py#L101

I was wondering if you have any thoughts?

**Comment by lucidrains** [comment]
@foobar8675 Hey Matthew again! It is arbitrary whether you designate 0, 1 as real, fake or fake, real - as long as the discriminator is optimizing for the opposite. In my framework, I have chosen 0 as real and 1 as fake

It's the adversarial nature of the circuit that makes the disentanglement emerge!

**Comment by lucidrains** [comment]
@foobar8675 this non-saturating logistic loss looks interesting though, i'll have to look into it a bit more before i switch. hinge loss has worked fine for me, and in my experience, loss function have mattered very little to the final generation quality

**Comment by foobar8675** [comment]
Thank you for the explanation!

**Comment by Mishra1995** [comment]
@lucidrains I am using the same module to train on flowers dataset. But the loss for generator after a period of time, overshoots and then reduces and this cycle keeps on going. Any idea regarding this?  The scale for overshooting and undershooting is very high.
